tools/automatic_query_fixer/src/test/java/com/google/cloud/bigquery/utils/queryfixer/FixerTest.java

```diff
@@ -74,6 +72,45 @@ public void fixTableNotFound() {
     assertEquals(22, result.getErrorPosition().getColumn());
   }
 
+  @Test
+  public void fixUnrecognizedColumn() {
+    String query = "SELECT state From `bigquery-public-data.austin_311.311_request` LIMIT 10";
+    String message = "Unrecognized name: state; Did you mean status? at [1:8]";
+    BigQuerySqlError error = buildError(message);
+
+    IFixer fixer = fixerFactory.getFixer(query, error);
+    assertTrue(fixer instanceof UnrecognizedColumnFixer);
+
+    FixResult result = fixer.fix();
+    assertEquals(1, result.getOptions().size());
+    assertEquals(
+        "SELECT status From `bigquery-public-data.austin_311.311_request` LIMIT 10",
+        result.getOptions().get(0).getFixedQuery());
+
+    assertEquals(1, result.getErrorPosition().getRow());
+    assertEquals(8, result.getErrorPosition().getColumn());
+  }
+
+  @Test
+  public void fixFunctionNotFound() {
```

Comment:
We should add more test cases for each error type, consider using parameterized tests or file-based test. We need to test more complicated queries. 
Feel free to add TODO and add it later.

Reply:
Agree. TODO is added.